plugins/login-plugin/src/Components/AssetFlow/index.js

```diff
@@ -56,22 +57,46 @@ const AssetFlow = (props) => {
     close_button: buttonUrl = "",
   } = screenStyles;
 
-  const assetId = inPlayerAssetId({
-    payload: props.payload,
-    configuration: props.configuration,
-  });
-
   const [dataSource, setDataSource] = useState(null);
   const [assetLoading, setAssetLoading] = useState(false);
+  const [assetId, setAssetId] = useState(null);
```

Comment:
if assetId, assetLoading and datasorce dempend on each other then you should use useReducer instead of three useState :)